test/modules/mongodb_cluster.go

```diff
@@ -0,0 +1,126 @@
+package modules
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+	"testing"
+
+	aws_sdk "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/gruntwork-io/terratest/modules/aws"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	test_structure "github.com/gruntwork-io/terratest/modules/test-structure"
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/atlas-sdk/v20231001002/admin"
+)
+
+func DeployMongoDBCluster(t *testing.T, workingDir string) {
+	// Get the role arn from env
+	roleArn := os.Getenv("TF_VAR_mongodb_role_arn")
+	// Generate a random id
+	randomID := random.UniqueId()
+	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		TerraformDir: workingDir,
+		Vars: map[string]interface{}{
+			"mongodb_role_arn": roleArn,
+			"random_id":        randomID,
+		},
+	})
+
+	// Save the options so later test stages can use them
+	test_structure.SaveTerraformOptions(t, workingDir, terraformOptions)
+}
+
+func ValidateMongoDBCluster(t *testing.T, workingDir string) {
```

Comment:
Can we add a comment on the function to explain its purpose as well as define which ENVs need to be exported, namely MONGODB_SECRET_ARN